LogsHandler/src/test/java/bootTest/IntegrationTest.java

```diff
@@ -42,36 +36,110 @@ public static void beforeAllTests() {
     public void testEndToEnd() {
         String key = RandomStringUtils.random(15, false, true);
         String jsonObjectAsString = "{\"message\":\"" + key + "\"}";
+        CreateAccountRequest createAccountRequest = new CreateAccountRequest("fadi");
+
+        Account account = createAccount(createAccountRequest);
+        String token = account.getToken();
+
+        indexAndAssert(token, jsonObjectAsString);
+
+        await().atMost(Duration.ofSeconds(10)).untilAsserted(() -> searchAndAssertMessage(key, account.getToken()));
+    }
+    @Test
+    public void testTwoAccounts() {
+        String message1Key = RandomStringUtils.random(15, false, true);
+        String message2Key = RandomStringUtils.random(15, false, true);
+
+        String message1 = "{\"message\":\"" + message1Key + "\"}";
+        String message2 = "{\"message\":\"" + message2Key + "\"}";
 
-        Response postResponse = webTarget.path("/index")
+        CreateAccountRequest createAccount1Request = new CreateAccountRequest("Fadi1");//{'accountName':'fadi'}";
+        CreateAccountRequest createAccount2Request = new CreateAccountRequest("Fadi2");//{'accountName':'fadi'}";
+
+        Account account1 = createAccount(createAccount1Request);
+        Account account2 = createAccount(createAccount2Request);
+
+        String token1 = account1.getToken();
+        String token2 = account2.getToken();
+
+        indexAndAssert(token1, message1);
+        indexAndAssert(token2, message2);
+
+        await().atMost(Duration.ofSeconds(10)).untilAsserted(() -> searchAndAssertAccountLogs(message2Key, account1.getToken()));
+        await().atMost(Duration.ofSeconds(10)).untilAsserted(() -> searchAndAssertAccountLogs(message1Key, account2.getToken()));
```

Comment:
There is one flaw in this test - ask me when you reach this comment

Reply:
What do you mean by that?